ui/src/app/workflows/components/workflow-logs-viewer/workflow-logs-viewer.tsx

```diff
@@ -32,13 +33,28 @@ export const WorkflowLogsViewer = ({workflow, nodeId, initialPodName, container,
     const [error, setError] = useState<Error>();
     const [loaded, setLoaded] = useState(false);
     const [logsObservable, setLogsObservable] = useState<Observable<string>>();
+    const [showRawLog, setShowRawLog] = useState(false);
 
     useEffect(() => {
         setError(null);
         setLoaded(false);
         const source = services.workflows
             .getContainerLogs(workflow, podName, nodeId, selectedContainer, grep, archived)
-            .map(e => (!podName ? e.podName + ': ' : '') + e.content + '\n')
+            // extract message from LogEntry
+            .map(e => {
+                let message = e.content;
+                if (!showRawLog) {
+                    try {
+                        const json = JSON.parse(message);
+                        if (json.message) {
+                            message = typeof json.message === 'string' ? json.message : JSON.stringify(json.message);
```

Comment:
Is `"message"` a standard field found across all structured logs? We would not want to over-fit a particular log format and hard-code it to our UI.

Reply:
> Is "message" a standard field found across all structured logs

I think  `message` is common field for log management. 

- [GCP structured-logging](https://cloud.google.com/logging/docs/structured-logging)
  - `The message that appears on the log entry line in the Logs Explorer.`
- [datadog](https://docs.datadoghq.com/logs/log_configuration/attributes_naming_convention/)
  - `By default, Datadog ingests the value of the message attribute as the body of the log entry. That value is then highlighted and displayed in Live Tail, where it is indexed for full text search.`

But as you know, sometimes this field is used as `msg`, `content`, and so on.
So I agree with you, it should be better to able to customize which fields user want.

I think this PR is first step to show json log easily. If it is better to add more features to merge, I will try.

Reply:
@simster7 Based on your review, I add more customizable ui on a log viewer and update PR description.